Sources/Danger/GitDiff.swift

```diff
@@ -0,0 +1,184 @@
+                return nil
+            }
+
+            return FileDiff(parsedHeader: parseHeader(header), hunks: parseHunks(Array(headerAndHunks.dropFirst())))
+        }
+    }
+
+    private func parseHeader(_ header: String) -> FileDiff.ParsedHeader {
+        let lines = header.components(separatedBy: "\n")
+
+        let filePath = lines.first?.split(separator: " ").first(where: { $0.starts(with: "b/") })?
+            .deletingPrefix("b/") ?? ""
+
+        let change: FileDiff.ParsedHeader.ChangeType
+
+        if lines.contains(where: { $0.hasPrefix("deleted file mode ") }) {
+            change = .deleted
+        } else if lines.contains(where: { $0.hasPrefix("new file mode") }) {
+            change = .created
+        } else if let modifiedLineIndex = lines.firstIndex(where: { $0.hasPrefix("rename from ") }) {
+            change = .renamed(oldPath: lines[modifiedLineIndex].deletingPrefix("rename from "))
+        } else {
+            change = .modified
+        }
+
+        return FileDiff.ParsedHeader(filePath: filePath, change: change)
+    }
+
+    private func parseHunks(_ hunks: [String]) -> [FileDiff.Hunk] {
+        (0 ..< hunks.count / 2).compactMap { index -> FileDiff.Hunk? in
+            let changesSpan = hunks[index * 2]
+            let changes = hunks[index * 2 + 1]
+            let lines = changes.components(separatedBy: "\n").dropFirst().dropLast().map(parseLine)
+            let parsedChanges = parseChangesSpan(changesSpan)
+
+            return FileDiff.Hunk(oldLineStart: parsedChanges.oldLineStart,
+                                 oldLineSpan: parsedChanges.oldLineSpan,
+                                 newLineStart: parsedChanges.newLineStart,
+                                 newLineSpan: parsedChanges.newLineSpan,
+                                 lines: lines)
+        }
+    }
+
+    private func parseLine(_ line: String) -> FileDiff.Line {
+        let text = String(line.dropFirst())
+        if line.hasPrefix("+") {
+            return FileDiff.Line(text: String(line.dropFirst()), changeType: .added)
+        } else if line.hasPrefix("-") {
+            return FileDiff.Line(text: String(line.dropFirst()), changeType: .removed)
+        } else {
+            return FileDiff.Line(text: text, changeType: .unchanged)
+        }
+    }
+
+    private func parseChangesSpan(_ changesSpan: String) -> HunkSpan {
+        let dividedSpan = changesSpan.split(separator: " ").map { $0.dropFirst().components(separatedBy: ",") }
+        if dividedSpan.count == 2,
+            dividedSpan[0].count == 2,
+            dividedSpan[1].count == 2,
+            let oldLineStart = Int(dividedSpan[0][0]),
+            let oldLineSpan = Int(dividedSpan[0][1]),
+            let newLineStart = Int(dividedSpan[1][0]),
+            let newLineSpan = Int(dividedSpan[1][1]) {
+            return HunkSpan(oldLineStart: oldLineStart, oldLineSpan: oldLineSpan, newLineStart: newLineStart, newLineSpan: newLineSpan)
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/GitDiff.swift;
  Line: 170;
-->

- :warning: Line should be 120 characters or less: currently 135 characters (`line_length`)